elastalert/ruletypes.py

```diff
@@ -1019,6 +1019,30 @@ def unwrap_term_buckets(self, timestamp, term_buckets):
     def check_matches(self, timestamp, query_key, aggregation_data):
         raise NotImplementedError()
 
+class ErrorRateRule(BaseAggregationRule):
+    """ A rule that determines error rate with sampling rate"""
+    required_options = frozenset(['sampling', 'threshold'])
+    def __init__(self, *args):
+        super(ErrorRateRule, self).__init__(*args)
+        self.ts_field = self.rules.get('timestamp_field', '@timestamp')
+        self.rules['metric_agg_key'] = "1"
+        self.rules['metric_agg_type'] = "count"
```

Comment:
We need to have cardinality query when finding total traces and just count query for error_traces? So metric_agg_type should be cardinality. But it's going as count only by default.